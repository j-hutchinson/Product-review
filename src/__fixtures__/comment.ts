import { Comment } from '../types';

export const singleComment: Comment = {
	name: 'Jack',
	email: 'something@aol.com',
	rating: 2,
	comment: 'adequate product'
};
