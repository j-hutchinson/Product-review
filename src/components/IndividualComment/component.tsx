import React from 'react';
import styled from 'styled-components';
import { Comment } from '../../types';

interface Props {
    comment: Comment
}

const StyledComment = styled.div`
    padding: 20px;
`;

const StyledLabel = styled.div``;

const IndividualComment = ({ comment }: Props) => {
    return (
        <StyledComment>
            <StyledLabel>Name: {comment.name}</StyledLabel>
            <StyledLabel>Rating: {comment.rating}</StyledLabel>
            <StyledLabel>Email: {comment.email}</StyledLabel>
            <StyledLabel>Comment: {comment.comment}</StyledLabel>
        </StyledComment>
    )
}

export default IndividualComment;