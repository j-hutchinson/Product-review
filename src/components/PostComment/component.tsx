import React, { ChangeEvent, FormEvent, useState } from 'react'
import Rater from 'react-rater';
import 'react-rater/lib/react-rater.css';
import styled from 'styled-components'
import { addComment } from '../../store/actions/actions';
import store from '../../store/store';

export const StyledContainer = styled.form``;

const StyledItem = styled.div`
    display: grid;
    grid-template-rows: min-content 1fr;
    grid-gap: 4px;
    margin: 10px;
`;

export const StyledInput = styled.input`
   height: 24px;
   font-size: 14px;
`;

const StyledButtonWrapper = styled.div`
    display: flex;
    flex-direction: row-reverse;
    margin-right: 10px;
`;

const StyledSubmitInput = styled.input`
    background-color: #03fc28;
    border-radius: 4px;
    border: 1px solid black;
    font-size: 14px; 
    padding: 8px;
`;

const StyledRatingLabel = styled.p`
    padding-left: 20px;
    display: inline-block;
`;

const PostComment = (): JSX.Element => {
    const [name, setName] = useState<string>('');
    const [email, setEmail] = useState<string>('');
    const [rating, setRating] = useState<number>(0);
    const [comment, setComment] = useState<string>('');

    const onSubmission = (ev: FormEvent): void => {
        ev.preventDefault();
        store.dispatch(addComment({ name, email, rating, comment }));
        setName('');
        setEmail('');
        setRating(0);
        setComment('');
    }

    return (
        <StyledContainer onSubmit={onSubmission}>
            <StyledItem>
                <label htmlFor="name"> Name: </label>
                <StyledInput
                    id="name"
                    type="text"
                    value={name}
                    onChange={(ev: ChangeEvent<HTMLInputElement>): void => setName(ev.target.value)}
                    required
                />
            </StyledItem>
            <StyledItem>
                <label htmlFor="email"> Email: </label>
                <StyledInput
                    id="email"
                    type="email"
                    value={email}
                    onChange={(ev: ChangeEvent<HTMLInputElement>): void => setEmail(ev.target.value)}
                    required
                />
            </StyledItem>
            <StyledItem>Rating:
                <div>
                    <Rater
                        rating={rating}
                        total={5}
                        onRate={({ rating }): void => setRating(rating)}
                    />
                    <StyledRatingLabel>({rating}) stars</StyledRatingLabel>
                </div>
            </StyledItem>
            <StyledItem>
                <label htmlFor="comment"> Comment </label>
                <StyledInput
                    id="comment"
                    type="text"
                    value={comment}
                    onChange={(ev: ChangeEvent<HTMLInputElement>): void => setComment(ev.target.value)}
                    required
                />
            </StyledItem>
            <StyledButtonWrapper>
                <StyledSubmitInput type="submit" value="Submit" />
            </StyledButtonWrapper>
        </StyledContainer>
    );
}
export default PostComment