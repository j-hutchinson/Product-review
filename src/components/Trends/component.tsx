import React from 'react';

const Trends = () => <div>graph</div>

export default Trends