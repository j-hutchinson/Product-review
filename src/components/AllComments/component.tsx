import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { Comment, ReduxState } from '../../types';
import IndividualComment from '../IndividualComment/component';

interface Props {
    allComments: Comment[]
}

const StyledCommentsContainer = styled.div`
    border: 1px solid black;
    border-radius: 4px;
    padding: 20px;
`;

export const AllComments = ({ allComments }: Props): JSX.Element => (
    <StyledCommentsContainer>
        <strong>Comments:</strong>
        {allComments.map((comment: Comment, ind: number): JSX.Element =>
            <IndividualComment comment={comment} key={`${comment.name}${ind}`} />)
        }
    </StyledCommentsContainer>
);

export const mapStateToProps = (state: ReduxState): Props => ({
    allComments: state.allComments
})

export default connect(mapStateToProps, null)(AllComments);