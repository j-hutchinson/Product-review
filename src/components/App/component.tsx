import React from 'react';
import styled from 'styled-components';
import AllComments from '../AllComments/component';
import PostComment from '../PostComment/component';

const StyledWrapper = styled.div`
  display: grid;
  grid-template-rows: 1fr 1fr;
  grid-gap: 40px;
  font-size: 14px;
  margin: auto;
  padding: 60px;
`;


const StyledSection = styled.div` 
  display: grid;
  grid-template-columns: 1fr 1fr;
`;

const App = (): JSX.Element => (
  <StyledWrapper>
    <StyledSection>
      <PostComment />
      <div>hello</div>
    </StyledSection>
    <AllComments />
  </StyledWrapper>
);

export default App;
