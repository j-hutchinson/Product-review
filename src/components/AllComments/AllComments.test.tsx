import { shallow } from 'enzyme';
import React from 'react';
import { AllComments, mapStateToProps } from './component';
import IndividualComment from '../IndividualComment/component';
import { ReduxState } from '../../types'
import { singleComment } from '../../__fixtures__/comment'

describe('AllComments component', () => {
    test('component matches snapshot with 1 comment', () => {
        expect.assertions(2);

        const wrapper = shallow(<AllComments allComments={[singleComment]} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(IndividualComment)).toHaveLength(1);
    });

    test('returns correct `props`', () => {
        expect.assertions(1);

        const state = {
            allComments: [singleComment]
        } as ReduxState;

        const props = mapStateToProps(state);

        expect(props).toEqual({ allComments: [singleComment] });
    });
});