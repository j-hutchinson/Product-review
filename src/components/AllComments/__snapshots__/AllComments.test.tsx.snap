// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllComments component component matches snapshot with 1 comment 1`] = `
<styled.div>
  <strong>
    Comments:
  </strong>
  <IndividualComment
    comment={
      Object {
        "comment": "adequate product",
        "email": "something@aol.com",
        "name": "Jack",
        "rating": 2,
      }
    }
    key="Jack0"
  />
</styled.div>
`;
